doc/documentation_workflow.rst

```diff
@@ -0,0 +1,154 @@
+Documentation workflow
+######################
+
+What you need to know
++++++++++++++++++++++
+
+We use `Sphinx <https://www.sphinx-doc.org/en/master/>`_ to generate documentation and `Read the Docs <https://readthedocs.org/>`_ to publish it.
+
+The NEST simulator documentation lives alongside its code. It is contained in the ``doc`` directory within the `NEST source code repository <https://github.com/nest/nest-simulator>`_ on GitHub.
+
+We work with `GitHub <https://www.github.com>`_ as the web-based hosting service for Git. Git allows us to keep our versions under control, with each release of NEST having its own documentation.
+
+.. image:: _static/img/documentation_workflow.png
+  :width: 300
+  :alt: Alternative text
+
+Changing the documentation
+++++++++++++++++++++++++++
+
+If you notice any errors or weaknesses in the documentation, please submit an `issue <https://github.com/nest/nest-simulator/issues>`_ in our GitHub repository.
+
+You can also make changes directly to your forked copy of the `NEST source code repository <https://github.com/nest/nest-simulator>`_ and create a `pull request <https://github.com/nest/nest-simulator/pulls>`_. Just follow the workflow below!
+
+Setting up your environment
++++++++++++++++++++++++++++
+
+To keep things simple, we have created a conda environment for you. Installing it will enable you to smoothly generate documentation for NEST.
+
+If you are using Linux and want to install a full development environment:
+
+1. Install conda (we recommend `miniconda <https://docs.conda.io/en/latest/miniconda.html#>`_).
+
+2. Switch to the ``doc`` folder in the source directory:
+
+.. code-block:: bash
+
+    cd </path/to/nest_source>/doc
+
+3. Create and activate the environment:
+
+.. code-block:: bash
+
+   conda env create -f nest_doc_conda_env.yml
+   conda update -n base -c defaults conda
+   conda activate nest-doc
+
+4. If you want to deactivate or delete the build environment:
+
+.. code-block:: bash
+
+   conda deactivate
+   conda remove --name nest-doc --all
+
+Generating documentation with Sphinx
+++++++++++++++++++++++++++++++++++++
+
+Now that you activated your environment, you can generate HTML files using Sphinx.
+
+Sphinx uses reStructuredText. To learn more on how to use it, check out this `quick reference <https://docutils.sourceforge.io/docs/user/rst/quickref.html>`_.
+
+Editing and creating pages
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+1. You can edit and/or add ``.rst`` files in the ``doc`` directory using your editor of choice.
+
+2. If you create a new page, open ``index.rst`` in the ``doc`` directory and add the file name under ``.. toctree::``. This will ensure it appears on the NEST simulator documentation's table of contents.
```

Comment:
I have problems executing this step. In doc/index.rst, there is no .. toctree:: to which I can add the file name of my newly created .rst file. When I add .. toctree::my_new_page.rst myself, it does not add my new page to the table of contents. What am I missing here?

Reply:
@jasperalbers Good that you noticed that. The ..toctree:: is in ``contents.rst``. Just corrected it.